### PR TITLE
fix: restructure mesheryctl version error handling

### LINKED ISSUE #1788: [mesheryctl] Check for latest release available not working when Meshery server is unavailable
**Description**
When Meshery server is available, mesheryctl version will perform the latest release check:
```
➜  ~ mesheryctl version
Client Version: v0.4.15 	  GitSHA: 51ca056186a824ad1c2b1bf979b95bdc46978d34
Server Version: v0.4.17 	  GitSHA: 06abac7
Checking for latest version of Meshery....
v0.4.17 is the latest Meshery release.
```

`mesheryctl version` no longer checks for latest release available when Meshery server is unavailable.

```
➜  ~ mesheryctl version
Client Version: v0.4.15 	  GitSHA: 51ca056186a824ad1c2b1bf979b95bdc46978d34
Server Version: Unavailable 	  GitSHA: Unavailable
Error Occurred: Get http://localhost:9081/api/server/version: dial tcp [::1]:9081: connect: connection refused
Could not communicate with Meshery at http://localhost:9081/api/server/version
Ensure that Meshery is available.
 See Meshery Documentation (https://docs.meshery.io) for help.
```

**Expected Behavior**
`mesheryctl version` should check for latest release available even when Meshery server is unavailable.

### PR DESCRIPTION
**Pull request by kushthedude**
Signed-off-by: Kush Trivedi <kushthedude@gmail.com>

**Description**

This PR fixes #1788

**Notes for Reviewers**
- For each of the functions invoked in `mesheryctl version`, the same error was getting propagated whether it be due to invalid response, unmarshalling of data, etc, this PR makes the error handling proper.
- Moves checkVersion to defer, irrespective of error in the function we will always see the check for the latest version of mesheryctl.

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/version.go [hunk 1]
```diff
@@ -65,37 +53,54 @@ var versionCmd = &cobra.Command{
...
-		if requestErr(err, url) {
+		if err != nil {
+			logrus.Infof("Server Version: %v \t  GitSHA: %v", version.Build, version.CommitSHA)
+			logrus.Errorf("\nUnable to get request context: %v", err)
 			return
 		}
 
+		defer checkMesheryctlClientVersion(build)
 		client := &http.Client{}
 		resp, err := client.Do(req)
```
Comment (nitishm) [inline mesheryctl/internal/cli/root/version.go:76]: Nit-pick: As good practice always close the response body `defer resp.Body.Close()`

**Review by nitishm — DISMISSED** [review]

File: mesheryctl/internal/cli/root/version.go (hunk 1 above)
Comment (kushthedude) [reply mesheryctl/internal/cli/root/version.go:76]: done !

File: mesheryctl/internal/cli/root/version.go [hunk 2]
```diff
@@ -65,37 +53,57 @@ var versionCmd = &cobra.Command{
...
+			Build:     "unavailable",
+			CommitSHA: "unavailable",
+		}
+
 		logrus.Infof("Client Version: %v \t  GitSHA: %v", build, commitsha)
 
 		req, err := http.NewRequest("GET", fmt.Sprintf("%s/server/version", url), nil)
-		if requestErr(err, url) {
+		if err != nil {
+			logrus.Infof("Server Version: %v \t  GitSHA: %v", version.Build, version.CommitSHA)
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/version.go:65]: Will you rephrase to make this message more human-readable?

Comment (kushthedude) [reply mesheryctl/internal/cli/root/version.go:65]: More human readable? This is the error response which was earlier too 
```sh
Server Version: unavailable 	  GitSHA: unavailable

Unable to communicate with Meshery: Get "http://localhost:9081/api/server/version": dial tcp [::1]:9081: connect: connection refused
Ensure that Meshery is available, see Meshery Documentation (https://docs.meshery.io) for more help.
```

**Comment by leecalcote** [comment]
@kushthedude thanks for this update. You might have a look at this item - https://github.com/layer5io/meshery/pull/1791/conflicts

**Comment by kushthedude** [comment]
> @kushthedude thanks for this update. You might have a look at this item - https://github.com/layer5io/meshery/pull/1791/conflicts

Done, also updated the error message in a version check which was pointing to meshery rather than at a working internet connection to check for GitHub release.

File: mesheryctl/internal/cli/root/version.go (hunk 2 above)
Comment (leecalcote) [reply mesheryctl/internal/cli/root/version.go:65]: Thank you, sir. Let's move forward with this.

**Review by leecalcote — APPROVED** [review]

**Pull request merged**